src/execution.ts

```diff
@@ -1178,14 +1179,19 @@ function specialValueReplacer(_: any, value: any) {
     return MAGIC_MINUS_INFINITY;
   }
 
+  if (this[key] instanceof Date) {
+    return MAGIC_DATE + this[key].getTime();
+  }
+
   return value;
 }
 
 function objectStringify(val: any): string {
   return JSON.stringify(val, specialValueReplacer)
-    .replace(`"${MAGIC_NAN}"`, "NaN")
-    .replace(`"${MAGIC_PLUS_INFINITY}"`, "Infinity")
-    .replace(`"${MAGIC_MINUS_INFINITY}"`, "-Infinity");
+    .replace(new RegExp(`"${MAGIC_NAN}"`, "g"), "NaN")
+    .replace(new RegExp(`"${MAGIC_PLUS_INFINITY}"`, "g"), "Infinity")
+    .replace(new RegExp(`"${MAGIC_MINUS_INFINITY}"`, "g"), "-Infinity")
```

Comment:
I noticed that these were only replacing the first occurrence of the magic strings, so I changed them to use a global regex.

Reply:
We could use replaceAll instead of making it a RegExp.

Reply:
replaceAll wasn't added to Node until v15, so you'd lose support for maintenance LTS versions.

Reply:
Ah. Cool!